src/storage/memory.rs

```diff
@@ -110,7 +110,8 @@ impl StorageInspect<ContractsRawCode> for MemoryStorage {
 
 impl StorageMutate<ContractsRawCode> for MemoryStorage {
     fn insert(&mut self, key: &ContractId, value: &[u8]) -> Result<Option<Contract>, Infallible> {
-        Ok(self.memory.contracts.insert(*key, value.clone().into()))
+        // Or Ok(self.memory.contracts.insert(*key, &(*value).clone().into())) if reference should be cloned
```

Comment:
I did have a question here, clippy was warning this is cloning a clone and not the underlying value, what would be desirable behavior here?

Reply:
`Ok(self.memory.contracts.insert(*key, value.into()))` is right

Reply:
So I haven't gotten around to fixing this one (will tomorrow), but I was wondering why it passes when I changed it clone the underlying value and not the clone of a clone? Shouldn't this break something or do our tests just not cover it?